PyInstaller/archive/writers.py

```diff
@@ -375,6 +376,26 @@ def add(self, entry):
 
                 code_data = marshal.dumps(code)
                 ulen = len(code_data)
+            elif typcd == 'm':
+                fh = open(pathnm, 'rb')
+                ulen = os.fstat(fh.fileno()).st_size
+                # Check if it is a PYC file
+                header = fh.read(4)
+                fh.seek(0)
+                if header == BYTECODE_MAGIC:
+                    # Read whole header and load code
+                    if is_py37:
+                        header = fh.read(16)
+                    else:
+                        header = fh.read(12)
```

Comment:
Do you have a link you can comment in to say where the 16 and the 12 come from?

Reply:
I've added a comment about header sizes and reference to PEP-552. 

And also a comment on why original header can be re-used after we modify the code object.

Reply:
Perfect.